### PR TITLE
Refactor smooks integration tests

### PR DESCRIPTION
**Pull request by Manuri**

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksIntegrationTestCase.java [hunk 1]
```diff
@@ -19,23 +19,22 @@
...
-import java.io.IOException;
 import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 import javax.activation.DataHandler;
-import javax.xml.stream.XMLStreamException;
 
 public class SmooksIntegrationTestCase extends ESBIntegrationTest {
```
Comment (milindaperera) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksIntegrationTestCase.java]: Shall we add class level javadoc comment with test description

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksIntegrationTestCase.java [hunk 2]
```diff
@@ -64,28 +63,29 @@ public void close() throws Exception {
...
+    public void testSendingToSmooks() throws Exception {
+        String smooksResourceDir = getClass().getResource(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "synapseconfig"
+                        + File.separator + "smooks" + File.separator).getPath();
+        Path source = Paths.get(smooksResourceDir, "person.csv");
+        Path destination = Paths.get(smooksResourceDir + "test", "in", "person.csv");
+
+        Files.createDirectories(Paths.get(smooksResourceDir, "test", "in"));
+        Files.copy(source, destination, StandardCopyOption.REPLACE_EXISTING);
+        Thread.sleep(2000);
```
Comment (milindaperera) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksIntegrationTestCase.java]: if we cannot avoid Thread sleep here shall we add a comment mentioning "for what we are waiting for"

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java [hunk 3]
```diff
@@ -18,134 +18,96 @@
...
+                        + "<date>2008-01-01</date>\n\t</header>\n\t"
+                        + "<order>\n\t\t"
+                        + "<price>50.00</price>\n\t\t"
+                        + "<quantity>500</quantity>\n\t\t"
+                        + "<symbol>IBM</symbol>\n\t\t"
+                        + "<comment>REF 10053</comment>\n\t"
+                        + "</order>\n"
+                        + "</orders>"
+                        + "</soapenv:Body>"
+                        + "</soapenv:Envelope>"));
```
Comment (manoramahp) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java]: Assert message is missing here

Comment (Manuri) [reply integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java]: Fixed

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
+/*
+*Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (srsharon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java]: The licence needs to be updated

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java [hunk 5]
```diff
@@ -18,134 +18,101 @@
 package org.wso2.carbon.esb.mediator.test.smooks;
```
Comment (srsharon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java]: Need to update the licence year.

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java (hunk 4 above)
Comment (Manuri) [reply integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java]: Actually I renamed an existing class therefore I'm not sure if its correct to change the license year.

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java (hunk 5 above)
Comment (Manuri) [reply integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java]: This is an existing class therefore we should not change the year.

Comment (srsharon) [reply integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java]: If you have made a change on the file you have to change the year. 
The year should the the year of the last edit.

Comment (Manuri) [reply integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromLocalEntryTestCase.java]: AFAIK the purpose of including the year is to show which year the code originated :)

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            super.cleanup();
+            resourceAdminServiceClient = null;
+        }
+    }
+
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE
+})
+    @Test(groups = {"wso2.esb"}, description = "Sending a Large File To Smooks Mediator")
+    public void testSendingToSmooks() throws Exception {
+        String smooksResourceDir = getClass().getResource(
```
Comment (abeykoon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java]: Shall we break the line at a proper (readable) place?

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import javax.activation.DataHandler;
+
+/**
+ * This test case is for verifying the functionality of processing relatively large files(>16K TCP buffer size) with
```
Comment (abeykoon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java]: Could you please clarify what does this line means?

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+    private boolean isProxyDeployed = false;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+
+        super.init();
+        resourceAdminServiceClient = new ResourceAdminServiceClient
+                (contextUrls.getBackEndUrl(), context.getContextTenant().getContextUser().getUserName()
+, context.getContextTenant().getContextUser().getPassword());
```
Comment (abeykoon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java]: Let's break line at a proper place.

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    private void uploadResourcesToConfigRegistry() throws Exception {
+        resourceAdminServiceClient.addResource(
+                "/_system/config/smooks_config.xml", "application/xml", "xml files",
+                new DataHandler(new URL("file:///" + getClass().getResource(
+                        "/artifacts/ESB/synapseconfig/smooks/smooks_config.xml").getPath())));
+    }
+
+    private void addSmooksProxy() throws Exception {
+
+        addProxyService(AXIOMUtil.stringToOM("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```
Comment (abeykoon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksLargeFileProcessingTestCase.java]: Can we read this from a file?

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromConfigRegistryTestCase.java [hunk 10]
```diff
@@ -18,158 +18,108 @@
...
 
     @BeforeClass(alwaysRun = true)
-    public void init() throws Exception {
+    public void setEnvironment() throws Exception {
+
         super.init();
-        resourceAdminServiceStub = new ResourceAdminServiceClient(contextUrls.getBackEndUrl(), context.getContextTenant().getContextUser().getUserName()
-, context.getContextTenant().getContextUser().getPassword());
+        resourceAdminServiceClient = new ResourceAdminServiceClient
+                (contextUrls.getBackEndUrl(), context.getContextTenant().getContextUser().getUserName()
```
Comment (abeykoon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromConfigRegistryTestCase.java]: Let's break the lines in a readable way.

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromConfigRegistryTestCase.java [hunk 11]
```diff
@@ -18,158 +18,108 @@
...
+        resourceAdminServiceClient.addResource(
+                "/_system/config/smooks_config.xml", "application/xml", "xml files",
+                new DataHandler(new URL("file:///" + getClass().getResource(
+                        "/artifacts/ESB/synapseconfig/smooks/smooks_config.xml").getPath())));
+    }
 
-        deleteProxyService("smooksMediatorAtConfigRegTestProxy");
+    private void addSmooksProxy() throws Exception {
+        addProxyService(AXIOMUtil.stringToOM("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                + "<proxy xmlns=\"http://ws.apache.org/ns/synapse\" name=\"SmooksProxy\" transports=\"vfs\" "
```
Comment (abeykoon) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/smooks/SmooksMediatorConfigFromConfigRegistryTestCase.java]: Can we read this from a file?

**Pull request merged**